js/app.js

```diff
@@ -0,0 +1,182 @@
+const app = {
+    appName: 'Race Canvas Lab',
+    version: '1.0.0',
+    license: undefined,
+    author: 'Pedro Suárez',
+    description: 'First Canvas project',
+    ctx: undefined,
+    imageInstance: undefined,
+    obs: [],
+    canvasSize: {
+        w: 500, h: 700,
+    },
+    carData: {
+        pos: { x: 200, y: 500 },
+        size: { w: 100, h: 175 },
+        image: './images/car.png'
+    },
+
+    framesCounter: 0,
+    score: 0,
+
+    init() {
+
+        this.setContext()
+        this.drawRoad()
+        this.createCar()
+        this.setEventHandlers()
+        this.createObs()
+        this.start()
+
+    },
+    setContext() {
+
+        this.ctx = document.querySelector('#canvas').getContext('2d');
+        console.log(this.ctx)
+
+    },
+
+    setEventHandlers() {
+        document.onkeydown = event => {
+            switch (event.key) {
+                case 'a':
+                    this.carData.pos.x -= 20
+                    break;
+                case 'ArrowLeft':
+                    this.carData.pos.x -= 20
+                    break;
+                case 'd':
+                    this.carData.pos.x += 20
+                    break;
+                case 'ArrowRight':
+                    this.carData.pos.x += 20
+                    break;
+                case 'w':
+                    this.carData.pos.y -= 20
+                    break;
+                case 's':
+                    this.carData.pos.y += 20
+            }
+        }
+    },
+
+    drawRoad() {
+
+        this.ctx.fillStyle = 'green'
+        this.ctx.fillRect(0, 0, this.canvasSize.w, this.canvasSize.h)
+        this.ctx.fillStyle = 'grey'
+        this.ctx.fillRect(50, 0, this.canvasSize.w - 100, this.canvasSize.h)
+
+
+
+        this.ctx.fillStyle = 'white'
+        this.ctx.fillRect(65, 0, 15, this.canvasSize.h)
+        this.ctx.fillStyle = 'white'
+        this.ctx.fillRect(this.canvasSize.w - 80, 0, 15, this.canvasSize.h)
+
+
+        this.ctx.moveTo(this.canvasSize.w / 2, 0)
+        this.ctx.lineTo(this.canvasSize.w / 2, this.canvasSize.h)
+        this.ctx.strokeStyle = 'white'
+        this.ctx.lineWidth = 10
+        this.ctx.setLineDash([60, 50])
+        this.ctx.stroke()
+
+    },
+
+    createCar() {
+        this.imageInstance = new Image()
+        this.imageInstance.src = this.carData.image
+    },
+
+    createObs() {
+        this.obs.push(
+            new Obstacle(this.ctx, this.canvasSize)
+        )
+    },
+
+    start() {
+        setInterval(() => {
```

Comment:
Aquí habría creado arriba interval : undefined
habria guardado el setInterbal ahi : this.interval = setInterval....
y por ultimo en el clearInterval puedes utilizar esa variable